griptape/drivers/image_generation/openai_image_generation_driver.py

```diff
@@ -39,42 +41,96 @@ class OpenAiImageGenerationDriver(BaseImageGenerationDriver):
+    )
+    quality: Optional[Literal["standard", "hd", "low", "medium", "high", "auto"]] = field(
+        default=None,
         kw_only=True,
         metadata={"serializable": True},
     )
-    image_size: Literal["256x256", "512x512", "1024x1024", "1024x1792", "1792x1024"] = field(
-        default="1024x1024", kw_only=True, metadata={"serializable": True}
+    image_size: Optional[Literal["256x256", "512x512", "1024x1024", "1024x1792", "1792x1024"]] = field(
+        default=None,
+        kw_only=True,
+        metadata={"serializable": True, "model_allowlist": ["dall-e-3"]},
+    )
+    response_format: Literal["b64_json"] = field(
+        default="b64_json",
+        kw_only=True,
+        metadata={"serializable": True, "model_denylist": ["gpt-image-1"]},
+    )
+    background: Optional[Literal["transparent", "opaque", "auto"]] = field(
+        default=None,
+        kw_only=True,
+        metadata={"serializable": True, "model_allowlist": ["gpt-image-1"]},
+    )
+    moderation: Optional[Literal["low", "auto"]] = field(
+        default=None,
+        kw_only=True,
+        metadata={"serializable": True, "model_allowlist": ["gpt-image-1"]},
+    )
+    output_compression: Optional[int] = field(
+        default=None,
+        kw_only=True,
+        metadata={"serializable": True, "model_allowlist": ["gpt-image-1"]},
     )
-    response_format: Literal["b64_json"] = field(default="b64_json", kw_only=True, metadata={"serializable": True})
+    output_format: Optional[Literal["png", "jpeg"]] = field(
+        default=None,
+        kw_only=True,
+        metadata={"serializable": True, "model_allowlist": ["gpt-image-1"]},
+    )
+    image_loader: ImageLoader = field(default=Factory(lambda: ImageLoader()), kw_only=True)
     _client: Optional[openai.OpenAI] = field(
         default=None, kw_only=True, alias="client", metadata={"serializable": False}
     )
 
+    @image_size.validator  # pyright: ignore[reportAttributeAccessIssue, reportOptionalMemberAccess]
+    def validate_image_size(self, attribute: str, value: str) -> None:
+        """Validates the image size based on the model.
+
+        Must be one of `1024x1024`, `1536x1024` (landscape), `1024x1536` (portrait), or `auto` (default value) for
+        `gpt-image-1`, one of `256x256`, `512x512`, or `1024x1024` for `dall-e-2`, and
+        one of `1024x1024`, `1792x1024`, or `1024x1792` for `dall-e-3`.
+
+        """
+        if value is None:
+            return
+
+        if self.model.startswith("gpt-image"):
+            allowed_sizes = ("1024x1024", "1536x1024", "1024x1536", "auto")
+        elif self.model == "dall-e-2":
+            allowed_sizes = ("256x256", "512x512", "1024x1024")
+        elif self.model == "dall-e-3":
+            allowed_sizes = ("1024x1024", "1792x1024", "1024x1792")
+        else:
+            raise NotImplementedError(f"Image size validation not implemented for model {self.model}")
```

Comment:
i think we should skip the `if gpt-image` check and just assume thats the default?
```suggestion
        if self.model == "dall-e-2":
            allowed_sizes = ("256x256", "512x512", "1024x1024")
        elif self.model == "dall-e-3":
            allowed_sizes = ("1024x1024", "1792x1024", "1024x1792")
        else:
            allowed_sizes = ("1024x1024", "1536x1024", "1024x1536", "auto")
```